test/unit/vl-proza-message.test.html

```diff
@@ -139,6 +155,175 @@
+        const message = fixture('vl-proza-message-fixture');
+        let typography = prozaTypographyElement(message);
+        let wysiwyg = prozaWysiwygElement(message);
+        should.not.exist(typography);
+        should.exist(wysiwyg);
+        return fetchMock.flush(true).then(() => {
+          typography = prozaTypographyElement(message);
+          wysiwyg = prozaWysiwygElement(typography);
+          should.exist(typography);
+          should.exist(wysiwyg);
+          prozaMessageEditButton(message).click();
+          typography = prozaTypographyElement(message);
+          wysiwyg = prozaWysiwygElement(message);
+          should.not.exist(typography);
+          should.exist(wysiwyg);
+        });
+      });
+
+      test('bij het klikken op de bewerk knop zal de bewerk modus geactiveerd worden', () => {
+        sandbox.spy(tinyMCE, 'init');
+        const message = fixture('vl-proza-message-fixture');
+        return fetchMock.flush(true).then(() => {
+          prozaMessageEditButton(message).click();
+          assert(tinyMCE.init.called);
+        });
+      });
+
+      test('bij het klikken op de bewerk knop zal na het activeren van de bewerk modus de bewerk knop verwijderd worden', (done) => {
+        sandbox.spy(tinyMCE, 'init');
+        const message = fixture('vl-proza-message-fixture');
+        fetchMock.flush(true).then(() => {
+          prozaMessageEditButton(message).click();
+          const editor = tinyMCE.activeEditor;
+          editor.on('init', () => {
+            should.not.exist(prozaMessageEditButton(message));
+            editor.destroy();
+            done();
+          });
+        });
+      });
+
+      test('bij het klikken op de bewerk knop krijgt de geactiveerde wysiwyg focus', (done) => {
+        sandbox.spy(tinyMCE, 'init');
+        const message = fixture('vl-proza-message-fixture');
+        fetchMock.flush(true).then(() => {
+          prozaMessageEditButton(message).click();
+          const editor = tinyMCE.activeEditor;
+          sandbox.spy(editor, 'focus');
+          editor.on('init', () => {
+            assert(editor.focus.called);
+            editor.destroy();
+            done();
+          });
+        });
+      });
+
+      test('bij het klikken op de bewerk knop krijgt de geactiveerde wysiwyg de typography class zodat in de bewerk modus zodat de wysiwyg stijl toegepast wordt', (done) => {
+        sandbox.spy(tinyMCE, 'init');
+        const message = fixture('vl-proza-message-fixture');
+        fetchMock.flush(true).then(() => {
+          prozaMessageEditButton(message).click();
+          const editor = tinyMCE.activeEditor;
+          editor.on('init', () => {
+            const wysiwyg = prozaWysiwygElement(message);
+            wysiwyg.classList.contains('vl-typography').should.to.be.true;
+            editor.destroy();
+            done();
```

Comment:
misschien een hulp-methode maken om al de "bij het klikken op de bewerk-knop ..." niet te veel duplicatie te hebben?

Reply:
Alleen ```prozaMessageEditButton(message).click();``` is dan duplicatie? want de code daarna is verschillend per test afhankelijk of er nog een spy moet gebeuren of niet. Dan zouden er veel argumenten meegegeven moeten worden aan de generieke functie en dat zorg uiteindelijk voor minder leesbaarheid.

Reply:
nee, t was de hele structuur van de testen volgens mij, maar niet belangrijk genoeg.